config/src/config/mod.rs

```diff
@@ -435,6 +454,36 @@ mod test {
         }
     }
 
+    #[test]
+    fn verify_get_waypoint() {
+        let node_config = NodeConfig::default_for_public_full_node();
+        let parsed_waypoint = Waypoint::from_str(
+            "0:01234567890ABCDEFFEDCA098765421001234567890ABCDEFFEDCA0987654210",
+        )
+        .unwrap();
+        assert_eq!(node_config.base.waypoint.waypoint(), parsed_waypoint);
+    }
```

Comment:
Looks good, I would recommended perhaps using a temporary on disk storage since this doesn't actually verify the path that is important for validators, namely, that genesis is read from the storage using the genesis key.

We can produce a temporary on disk storage:
https://github.com/diem/diem/blob/master/secure/storage/src/tests/on_disk.rs#L9-L10
And pass that in via https://github.com/diem/diem/blob/master/config/src/config/secure_backend_config.rs#L70

So you could test with it set in storage and without ensuring full compatibility.